pkg/gadgets/trace/bind/tracer/tracer.go

```
@@ -276,6 +263,36 @@ func (t *Tracer) run() {
 			t.enricher.EnrichByMntNs(&event.CommonData, event.MountNsID)
 		}
 
+		interfaceString := ""
+
+		interfaceNum := int(bpfEvent.BoundDevIf)
+		event.Index = interfaceNum // for debugging purposes. will be removed.
+		if interfaceNum != 0 {
+			// It does exist a net link which index is 0.
+			// But eBPF bindsnoop code often gives 0 as interface number:
+			// https://github.com/iovisor/bcc/blob/63618552f81a2631990eff59fd7460802c58c30b/tools/bindsnoop_example.txt#L16
+			// So, we only use this function if interface number is different than 0.
+			var interf netlink.Link
+			CRPID := int(event.CommonData.Runtime.ContainerPID)
+			err := nsenter.NetnsEnter(CRPID, func() error {
+				interf, err = netlink.LinkByIndex(interfaceNum)
+				if err != nil {
+					msg := fmt.Sprintf("Cannot find interface for index %d : %s", interf, err)
+					t.eventCallback(types.Base(eventtypes.Err(msg)))
+					return nil
```

All this logic is running inside a function, so instead of calling t.eventCallback, it should return an error here, then the conditional on L286 will catch it and trigger the t.eventCallback.